src/tests/test_badges.py

```diff
@@ -22,6 +23,17 @@ def test_create_badge(self):
         created_badge = BadgeFactory()
         assert created_badge.id
 
+    def test_volunteer_badge_finding(self, user_client):
+        """Тестирование фильтрации волонтеров по бэйджу."""
+        badge_1, badge_2 = BadgeFactory(), BadgeFactory()
+        user_1, user_2 = UserFactory(), UserFactory()
+        _, _ = VolunteerBadgeFactory(user=user_1, badge=badge_1), VolunteerBadgeFactory(user=user_2, badge=badge_2)
```

Comment:
необходимые связи уже прописаны на уровне фабрики

```suggestion
        vb1, vb2 = VolunteerBadgeFactory(), VolunteerBadgeFactory()
```